staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -146,7 +146,7 @@ func NewCmdExposeService(f cmdutil.Factory, streams genericclioptions.IOStreams)
 	cmd.Flags().String("generator", "service/v2", i18n.T("The name of the API generator to use. There are 2 generators: 'service/v1' and 'service/v2'. The only difference between them is that service port in v1 is named 'default', while it is left unnamed in v2. Default is 'service/v2'."))
 	cmd.Flags().String("protocol", "", i18n.T("The network protocol for the service to be created. Default is 'TCP'."))
 	cmd.Flags().String("port", "", i18n.T("The port that the service should serve on. Copied from the resource being exposed, if unspecified"))
-	cmd.Flags().String("type", "", i18n.T("Type for this service: ClusterIP, NodePort, LoadBalancer, or ExternalName. Default is 'ClusterIP'."))
+	cmd.Flags().String("type", "", i18n.T("Type for this service: ClusterIP, NodePort, or LoadBalancer. Default is 'ClusterIP'."))
```

Comment:
@zhouya0 I've seen your comment from https://github.com/kubernetes/kubernetes/issues/87398#issuecomment-576561338 and that only proofs me more in my mission to get rid of generators. If we had the code responsible for creating that service baked in here we'd had full control over how the resulting service will look like and adding the support for ExternalName would be simple. I'd prefer we not limit functionality but move the code responsible for creating the service here. Now with `kubectl run` creating just pods we don't need to share the code and I'd rather see this fixed this way.

Reply:
Thanks @soltysh ! I had a long time think about this but not found the best solution, I will absolutly listen your advice. But I'n not sure I totally get your point:
```
but move the code responsible for creating the service here
```
From where to where?